src/main/java/com/puppycrawl/tools/checkstyle/meta/MetadataGeneratorUtil.java

```diff
@@ -74,8 +107,190 @@ public static void generate(String path, OutputStream out, String... moduleFolde
         checker.addListener(new MetadataGeneratorLogger(out, OutputStreamOptions.NONE));
 
         final List<File> moduleFiles = getTargetFiles(path, moduleFolders);
+        final List<File> moduleFilesForNewMetadata = moduleFiles.stream().filter(file -> {
+            final String fileName = file.getName();
+            return MODULES_GENERATED_BY_NEW_FUNCTIONALITY.contains(fileName);
+        }).collect(Collectors.toList());
+        moduleFiles.removeAll(moduleFilesForNewMetadata);
 
         checker.process(moduleFiles);
+
+        //////////////////////////////
+        // Plan:
+        // 1. Module name
+        //    - get from file name
+        //    - remove "Check"
+        // 2. Fully qualified name
+        //    clss.getName()
+        // 3. Parent
+        //    SiteUtil.getParentModule(clss)
+        // 4. ModuleType
+        //    - use getModuleType() from JavadocMetadataScraper
+        // 5. Description
+        //    - everything in javadoc is description
+        // 6. Violation message keys
+        //    - SiteUtil.getMessageKeys(clss)
+        //////////////////////////////
+        for (File file: moduleFilesForNewMetadata) {
+            final ModuleDetails moduleDetails = new ModuleDetails();
+
+            String moduleName = getModuleName(file);
+            final String checkModuleExtension = "Check";
+            if (moduleName.endsWith(checkModuleExtension)) {
+                moduleName = moduleName
+                        .substring(0, moduleName.length() - checkModuleExtension.length());
+            }
+            moduleDetails.setName(moduleName);
+
+            // TODO: Move to shared module, because that MacroExecutionException doesn't belong here
+            final Object instance = SiteUtil.getModuleInstance(moduleName);
+            final Class<?> clss = instance.getClass();
+            final String fullyQualifiedName = clss.getName();
+            moduleDetails.setFullQualifiedName(fullyQualifiedName);
```

Comment:
Instead of setters, can we use all-argumets constructor of module details?
And move all calculations of value to special methods.

So our code with look:
```
Final String name = getFQN(....)
Object parent = get parent(...)
.....
ModuleDetails moduleDetails = new ModuleDetails(name, parent,.....)
```

Same for ModulePropertyDetails.

Reply:
Done.